jwst/photom/photom.py

```diff
@@ -390,29 +354,21 @@ def calc_miri(self, ftab):
         if self.detector == 'MIRIMAGE':
 
             # Get the subarray value of the input data model
-            subarray = self.input.meta.subarray.name
+            subarray = self.input.meta.subarray.name.upper()
             log.info(' subarray: %s', subarray)
-
-            # Find the matching row in the reference file
-            for tabdata in ftab.phot_table:
-
-                ref_filter = tabdata['filter'].strip().upper()
-                ref_subarray = tabdata['subarray'].strip().upper()
-
-                # If the ref file subarray entry is GENERIC, it's an automatic
-                # match to the science data subarray value
-                if ref_subarray == 'GENERIC':
-                    ref_subarray = subarray
-
-                # Find matching FILTER and SUBARRAY values
-                if self.filter == ref_filter and subarray == ref_subarray:
-                    self.photom_io(tabdata)
-                    match = True
-                    break
-
-            if not match:
-                log.warning('No match in reference file')
-
+            fields_to_match = {'subarray': self.input.meta.subarray.name,
+                               'filter': self.filter}
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore")
```

Comment:
I replicated the previous implementation. It looks like the MIRI photom table allows subarray to be `GENERIC`, and it is used in case the subarray in the observation is not in the reference table. So the matching code makes one pass through the table looking for the subarray used in the observation. If not found it matches only filter and uses `GENERIC` as a subarray name.